src/General/Icon/Icon.test.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
+import * as renderer from 'react-test-renderer';
+import { fireEvent, render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import Icon from './Icon';
+
+const testId = 'icon-svg';
+const children = (
+  <path d="M100 56.563H56.562V100H43.438V56.562H0V43.438h43.438V0h13.124v43.438H100z"></path>
+);
+
+it('<Icon> should render a svg with color, width, height and a path as children', () => {
+  const IconSnapshot = renderer.create(<Icon>{children}</Icon>).toJSON();
+  expect(IconSnapshot).toMatchSnapshot();
+});
+
+it('<Icon> should accept a className to apply the style from styled-component', () => {
```

Comment:
why is `<Icon />` related to `styled-component`?

Reply:
ah, if you want to make sure it can be wrapped by `styled`, maybe you can do it in the test, otherwise I think the description can be `<Icon /> should accept a className prop`